packages/component-catalog-spa/src/Hooks/useFetchReadme.js

```diff
@@ -0,0 +1,53 @@
+import { useEffect, useState } from "react";
+
+const useFetchReadme = (componentLink) => {
+  const [data, setData] = useState(null);
+  const [isPending, setIsPending] = useState(true);
+  const [error, setError] = useState(null);
+  const [images, setImages] = useState([]);
+  const abortErrorSetter = (err) => {
+    if (err.name === 'AbortError') {
+      console.warn('Fetch Aborted!');
+    } else {
+      setIsPending(false);
+      setError(err.name);
+    }
+  };
+
+  useEffect(() => {
+    const abortController = new AbortController();
+    const signal = abortController.signal;
+    (async function () {
+      try {
+        const componentsDir = await fetch(componentLink, { signal });
```

Comment:
@hybridx did you check the graphql api provided by GitHub i think you can save a couple of calls using that

Reply:
@akhilmhdh I did look into the graphql queries https://gist.github.com/MichaelCurrin/6777b91e6374cdb5662b64b8249070ea and it seem a bit complicated comparing with REST

Reply:
But it could save 3  calls to just one

Reply:
I suppose we can add it in the next iteration